Refactor/module name

doronz88 opened the pull request.

rpdelaney commented on xontrib/branch_expansion.py:
```diff
@@ -32,7 +33,11 @@ def expand_braces(cmd, **_):
                 prefix + option + postfix for option in options.split(",")
             ]
             expanded_cmds.extend(expanded_parts)
+            requires_expansion = True
```
Why not just return here? No need to wait

doronz88 replied:
the `{}` can exist in more parts

rpdelaney reviewed (changes requested):
I don't see how this restructures the module name. Do you have unpushed changes?

doronz88 commented:
> I don't see how this restructures the module name. Do you have unpushed changes?

Renamed the python module to be imported via `xontrib load` (look at the first commit)

rpdelaney commented:
Ah, looks like you renamed it to `branch_expansion` but it should be `brace_expansion`

rpdelaney commented on xontrib/branch_expansion.py:
```diff
@@ -32,7 +33,11 @@ def expand_braces(cmd, **_):
                 prefix + option + postfix for option in options.split(",")
             ]
             expanded_cmds.extend(expanded_parts)
+            requires_expansion = True
```
```suggestion
            return " ".join(expanded_cmds)
```

doronz88 replied:
but that way i don't know if there are parts with both `{` and `}` so i can stop the expansion

rpdelaney commented on xontrib/branch_expansion.py:
```diff
@@ -32,7 +33,11 @@ def expand_braces(cmd, **_):
             expanded_cmds.append(part)
 
-    return " ".join(expanded_cmds)
+    if requires_expansion:
+        return " ".join(expanded_cmds)
+
+    return cmd
```
```suggestion
```

rpdelaney commented on xontrib/branch_expansion.py:
```diff
@@ -32,7 +33,11 @@ def expand_braces(cmd, **_):
             expanded_cmds.append(part)
 
-    return " ".join(expanded_cmds)
+    if requires_expansion:
+        return " ".join(expanded_cmds)
+
+    return cmd
```
```suggestion
```

doronz88 replied:
as i said previosly, this is needed

rpdelaney commented on xontrib/branch_expansion.py:
```diff
@@ -23,6 +23,7 @@
 def expand_braces(cmd, **_):
     parts = cmd.split()
     expanded_cmds = []
+    requires_expansion = False
```
```suggestion
```

doronz88 committed:
add `.gitignore`

doronz88 committed:
refactor module name to `brace_expansion`

This matches the expected behavior of a xontrib as well as the existing
documentation.

Close #1

doronz88 committed:
fix cmd expansion on more complicated commands

This fixes the issue occurring when expanding cmds such as:
```shell
echo .
echo @('a{1,2}')
```

rpdelaney commented on xontrib/brace_expansion.py:
```diff
@@ -21,8 +21,12 @@
 
 @events.on_transform_command
 def expand_braces(cmd, **_):
+    if '[' in cmd or '$(' in cmd or '@(' in cmd:
+        return cmd
+
```
Please don't sprawl the scope of the PR. I don't like this and won't accept it; whatever this is doing should be done with an ast, but more importantly, it should be another PR so we can discuss that separately.

rpdelaney commented on xontrib/brace_expansion.py:
```diff
@@ -21,8 +21,12 @@
+    requires_expansion = False
 
     for part in parts:
         if "{" in part and "}" in part:
```
A regular expression might be more reliable and could exclude nested braces. Could be like this:

```python
import re

pattern = r"\{([a-zA-Z],?)+\}"
test_strings = ["{a,o,e,u}", "{x,y,z}", "{aoeu}", "{a,o,e,u,{x},y}"]

for test in test_strings:
    if re.match(pattern, test):
        print(f"Match: {test}")
    else:
        print(f"No match: {test}")
```

rpdelaney commented on xontrib/brace_expansion.py:
```diff
@@ -32,7 +36,11 @@ def expand_braces(cmd, **_):
+    if requires_expansion:
+        return " ".join(expanded_cmds)
+
+    return cmd
```
If we reach line 39, we always end up returning from line 44. There is no reachable code between these two points. Please remove `requires_expansion`.

rpdelaney replied:
It would really be better to change one thing at a time to avoid having the core issue bogged down.
